applicationLibrary/molecularDynamics/molecularDynamicsLibrary/AxilrodTellerMutoFunctorHWY.h

```diff
@@ -0,0 +1,1836 @@
+              potentialEnergySum, numKernelCallsN3Sum, numGlobalCalcsN3Sum, numKernelCallsNoN3Sum,
+              numGlobalCalcsNoN3Sum);
+        }
+
+        const auto restK = soa3.size() - k;
+
+        if (restK > 0) {
+          handleKLoopBody<DistanceMatrix<false, alignedSoAView>, DistanceMatrix<false, alignedSoAView>,
+                          /*newton3*/ newton3, /*newton3Kernel*/ newton3,
+                          /*remainder*/ true, alignedSoAView>(
+              i, j, k, const_nu, interSoA2SoA3Dists, interSoA1SoA3Dists, ownedStatePtr3, typeptr1, typeptr2, typeptr3,
+              distXIJVec, distYIJVec, distZIJVec, distSquaredIJVec, invR5IJVec, fXAccI, fYAccI, fZAccI, fXAccJ, fYAccJ,
+              fZAccJ, ownedStateI, ownedStateJ, fxPtr3, fyPtr3, fzPtr3, virialSumX, virialSumY, virialSumZ,
+              potentialEnergySum, numKernelCallsN3Sum, numGlobalCalcsN3Sum, numKernelCallsNoN3Sum,
+              numGlobalCalcsNoN3Sum, restK);
+        }
+        // Reduce force on particle j.
+        fxPtr2[j] += highway::ReduceSum(tag_double, fXAccJ);
+        fyPtr2[j] += highway::ReduceSum(tag_double, fYAccJ);
+        fzPtr2[j] += highway::ReduceSum(tag_double, fZAccJ);
+      }
+      // Reduce force on particle i.
+      fxPtr1[i] += highway::ReduceSum(tag_double, fXAccI);
+      fyPtr1[i] += highway::ReduceSum(tag_double, fYAccI);
+      fzPtr1[i] += highway::ReduceSum(tag_double, fZAccI);
+    }
+    if constexpr (countFLOPs) {
+      _aosThreadDataFLOPs[threadnum].numTripletsCount += numTripletsCountingSum;
+      _aosThreadDataFLOPs[threadnum].numDistCalls += numDistanceCalculationSum;
+      _aosThreadDataFLOPs[threadnum].numKernelCallsNoN3 += numKernelCallsNoN3Sum;
+      _aosThreadDataFLOPs[threadnum].numKernelCallsN3 += numKernelCallsN3Sum;
+      _aosThreadDataFLOPs[threadnum].numGlobalCalcsNoN3 += numGlobalCalcsNoN3Sum;
+      _aosThreadDataFLOPs[threadnum].numGlobalCalcsN3 += numGlobalCalcsN3Sum;
+    }
+    if (calculateGlobals) {
+      _aosThreadDataGlobals[threadnum].potentialEnergySum += highway::ReduceSum(tag_double, potentialEnergySum);
+
+      _aosThreadDataGlobals[threadnum].virialSum[0] += highway::ReduceSum(tag_double, virialSumX);
+      _aosThreadDataGlobals[threadnum].virialSum[1] += highway::ReduceSum(tag_double, virialSumY);
+      _aosThreadDataGlobals[threadnum].virialSum[2] += highway::ReduceSum(tag_double, virialSumZ);
+    }
+  }
+
+  template <bool newton3>
+  void SoAFunctorVerletImpl(autopas::SoAView<SoAArraysType> soa, const size_t indexFirst,
+                            const std::vector<size_t, autopas::AlignedAllocator<size_t>> &neighborList) {
+    autopas::utils::ExceptionHandler::exception(
+        "AxilrodTellerMutoFunctorHWY::SoAFunctorVerletImpl() is not implemented.");
+  }
+
+  template <class DistanceMatrixT1, class DistanceMatrixT2, bool newton3, bool newton3Kernel, bool remainder,
```

Comment:
I'm reluctant about these as class templates. We know that these will always be `DistanceMatrix<???, alignedSoAView>`, where `???` is a bool. I would simply make this a `bool` template argument then.